Add team policy document (#176)
rust-lang/style-team: team-policy.md

# Style Team Policy Document

This is a living document tracking the active policies of the style team. It is intended to fill a similar role to the books that many[^1] teams[^2] maintain[^3] independently[^4] with the intent that eventually these can be integrated into a central policy repository for all teams in the rust-lang organization.

* **Original RFC**: https://rust-lang.github.io/rfcs/3309-style-team.html
* **Charter**: TODO

## Operational Roles

> The style team has various operational roles to delegate releated and ongoing work to specific individuals, to clarify on our operations, and to coordinate sharing feedback with the members in those roles to enable iterative self improvement.

Reviewer: Minor typo

```suggestion
The style team has various operational roles to delegate related and ongoing work to specific individuals, to clarify on our operations, and to coordinate sharing feedback with the members in those roles to enable iterative self improvement.
```

### Operational Lead

* Current Holder: [@calebcartwright](https://github.com/calebcartwright)
* Next Feedback Date: TODO

In order to stay in touch with where we want to be heading in the future, we need leadership. A team leader is paying attention to the team’s operations in relation to the team’s aim. What needs to be done, who agreed to do it. What is in the future to decide?

Responsibilities:

* Pays attention to the operation of the team
* Pays attention to the team members
* Reports wider context to the team

### Facilitator

* Current Holder: [@yaahc](https://github.com/yaahc)
* Next Feedback Date: Wednesday 2023-02-01

> In order to be present with each other, we need a good facilitator .  Facilitators run meetings according to the format of meetings and decision making adopted by the group. Leader and facilitator are separate roles because facilitation and overseeing operations are different skill sets. They can be held by the same individual.

Reviewer: Minor item, but I think we're only planning on having a single facilitator and that it would be good to use the singular form to avoid any potential misinterpretations of there being multiple 

```suggestion
In order to be present with each other, we need a good facilitator.  The facilitator runs meetings according to the format of meetings and decision making adopted by the group. Leader and facilitator are separate roles because facilitation and overseeing operations are different skill sets. They can be held by the same individual.
```

Responsibilities:

* Facilitates team meetings.
* Pays attention to equivalence[^5] during meetings.
* Supports planning of the agenda.
* Distinguishes facilitator voice from team member voice.

### Secretary

* Current Holder: [@calebcartwright](https://github.com/calebcartwright)
* Next Feedback Date: TODO

> In order to manage continuity with the team’s past, we need to have written records. The secretary manages the notes during the meeting, makes sure the minutes are distributed and accessible. The secretary also manages the records of the team and is the interpreter of policies.

Reviewer: The last bit of the last sentence gave me pause. Would you mind elaborating on what it means for the secretary to be the interpreter of policies?

Author: I'm not super attached to this framing and got it from other org's ways of conceptualizing their secretary roles. The idea as I understand it being that if there's an ambiguity and some of the team members have conflicting interpretation of said ambiguity it's helpful to have a singular person who is authorized to immediately resolve the ambiguity. I'm happy to leave this out and revisit it later if it crops up as an issue.

Responsibilities:

* Makes sure meeting minutes are taken, approved, and stored.
* Keeps track of all documents of the team.
* Interprets the meeting minutes in case of disagreement.
* Supports planning the agenda from the backlog.
    * Tracking when policies and roles are due for review.

## Meetings

> The style team meets weekly on wednesdays at 12:30am PST [(everytimezone link)](https://everytimezone.com/s/3f88a253) for a weekly video sync. Agendas are posted in the [#T-style] zulip stream. 

Reviewer 2: ```suggestion
The style team meets weekly on wednesdays at 12:30pm PST [(everytimezone link)](https://everytimezone.com/s/3f88a253) for a weekly video sync. Agendas are posted in the [#T-style] zulip stream. 
```

> Unless otherwise noted, all of our meetings are public and open for anyone to attend. You will find the timing and event details on our [style team calendar](https://calendar.google.com/calendar/embed?src=d0564ed914a41cf4915bd5ebe6e2e4ec0ee1293fdc1d09d6f5bdb27d4f91c083%40group.calendar.google.com&ctz=America%2FLos_Angeles). We publish notes and minutes in written form in this github repository.

Reviewer: Are we publishing minutes in this repo? Or is this implicitly covered via the link to the Hackmd?

Author: I was thinking it might be a good idea to publish minutes each time we hit the length limit on the hackmd and yea I was leaning towards this repo, tho I'm also planning on talking to other teams about centralizing project policy in one location and I feel like centralizing minutes would be a good thing to include in that initiative for similar reasons, so maybe not this repo long term.

### Agenda, Backlog, and Minutes

> The style team stores it's agendas, backlog, and meeting minutes in a single live [hackmd document](https://hackmd.io/@yaah/B1GZrzv4j) (rotating as necessary when the documents reach their length limit).

Reviewer: ```suggestion
The style team stores its agendas, backlog, and meeting minutes in a single live [hackmd document](https://hackmd.io/@yaah/B1GZrzv4j) (rotating as necessary when the documents reach their length limit).
```

> The agenda should be prepared in advanced of each meeting by the facilitator with the assistance of the secretary and the lead. The lead is responsible for looking forward and adding agenda items such as new requests from other teams and users or new priorities and goals. The secretary is responsible for looking backwards and adding agenda items from existing work such as items from the backlog or policies that are due for review. The agenda proposal is then presented at the beginning of each meeting by the facilitator for the rest of the team to consent or object too.

Reviewer: Not sure how literally we're intending to hold ourselves to this, but if we are planning on going word for word then I'd suggest we set an earlier mark for getting the agenda out. We've typically been running tight on time in past meetings and I think it'd be better to have the agenda set and generally agreed upon before the start of the meeting

Author: I tend to like to interpret things more literally. Happy to have the agenda proposal happen before meetings.

Reviewer 3: Proposed before meetings yes; I think it's important for the agenda-consent process to happen in the meeting, though.

#### Inform / Explore / Decide agenda item classification

Agenda items are labeled according to their desired outcome. There are three possible outcomes for any given agenda item, inform, explore, or decide. Each of these outcomes builds upon the previous outcomes. In order to explore an item it must first be understood, and in order to make a decision one must both understand and explore the item first. Inform is used for things such as status updates, and usually involves first a report, followed by a round of clarifying questions. Explore is used for situations where one would like feedback on a potential issue or proposal, and usually involves reaction rounds. Decide is used when the team must reach a decision as a group, and it is usually achieved via a consensus round.

> We identify the desired outcome in advanced to avoid aimless discussions with unclear goals. The desired outcome can change during discussion as new information becomes available. It is the facilitators responsibility to notice when this happens and formally make the change in the desired outcome with consent from the rest of the team. Extra clarity can be gained if, for every agenda item, we end by measuring whether we have achieved the desired outcome. Facilitators can make it a habit to pause before moving to a new agenda item by assessing whether the desired outcome has been achieved and by asking the secretary to read out loud what has been written in the notes.

Reviewer: ```suggestion
We identify the desired outcome in advance to avoid aimless discussions with unclear goals. The desired outcome can change during discussion as new information becomes available. It is the facilitator's responsibility to notice when this happens and formally make the change in the desired outcome with consent from the rest of the team. Extra clarity can be gained if, for every agenda item, we end by measuring whether we have achieved the desired outcome. Facilitators can make it a habit to pause before moving to a new agenda item by assessing whether the desired outcome has been achieved and by asking the secretary to read out loud what has been written in the notes.
```

* “What worked well in the meeting?”
* “What could be improved in future meetings?”
* “Is there anything you are carrying out of the meeting that you’d like to get off your chest now?”

Meeting evaluations are an opportunity to learn from our meetings. We can either talk about content, process, or interpersonal dynamics. We utilize meeting evaluations to help ourselves inhabit a growth mindset. Our goal is to have meetings which are well-run, refreshing, connecting, and energizing. We achieve this goal by giving space for people to speak up about both the positive and negative aspects of how we're working together so that we can prioritize continuous improvement and positive connections.

## Style Guide Evolution

The Rust style guide will generally match the latest version of the Rust style; the style team does not plan to maintain multiple branches of the style guide for different editions, in part because formatting for new constructs will apply to any edition supporting those constructs.

> Whenever possible, style decisions should be made before a new construct is stabilized. However, style decisions shall not be considered a blocker for stabilization.

Reviewer: What would you think about moving this section somewhere higher in the document, at least before the meeting minutes template?

I ask because I feel like it's one of the more critical and impactful decisions we've made and worry that it might get a little lost at the bottom following the meeting template (especially for those reading the non-rendered markdown)

# Style Team Policy Document

> This is a living document tracking the active policies of the style team. It is intended to fill a similar role to the books that many[^1] teams[^2] maintain[^3] independently[^4] with the intent that eventually these can be integrated into a central policy repository for all teams in the rust-lang organization.

Reviewer 3: ```suggestion
This is a living document tracking the active policies of the style team. It is intended to fill a similar role to the books that many[^1] teams[^2] maintain[^3] independently[^4]. If in the future Rust establishes central linking/indexing/aggregating of policies, these will need to appear there.
```

* **Original RFC**: https://rust-lang.github.io/rfcs/3309-style-team.html

> * **Charter**: TODO

Reviewer 3: We can directly link to charter.md from here.

## Style Guide Evolution

The Rust style guide will generally match the latest version of the Rust style; the style team does not plan to maintain multiple branches of the style guide for different editions, in part because formatting for new constructs will apply to any edition supporting those constructs.

Whenever possible, style decisions should be made before a new construct is stabilized. However, style decisions shall not be considered a blocker for stabilization.

> 

Author: ```suggestion

## Policy Review Process

To ensure that our policies remain up-to-date and effective, the style team implements a policy review process. This process involves setting review dates for all policies, documenting them alongside the policy itself, and including them as items in the backlog.

Review dates are selected based on the confidence level in a policy and its recent history of changes. If a policy has undergone recent modifications or is relatively new, a shorter review period is appropriate. This allows us to assess its impact and make any necessary adjustments promptly. On the other hand, if a policy has remained unchanged for an extended period and has proven to be effective, a longer review period can be chosen to minimize unnecessary energy expendeture.

By establishing review dates and incorporating them into our policy documentation and backlog, we ensure that policies are regularly evaluated and kept in line with the evolving needs of our team. This proactive approach help us keep our policies well-maintained and relevant.

```

Author: ```suggestion

## Policy Review Process

To maintain the effectiveness and relevance of our policies, the style team implements a policy review process. The following guidelines outline how we conduct policy reviews:

1. Associated Review Dates: Each policy should have a designated review date, which is documented alongside the policy itself. These review dates are also added as items in the team's backlog for easy tracking and planning.
2. Consideration Factors: When selecting review dates for policies, we take several factors into account. These include:
  1. Confidence Level: If a policy has recently undergone changes or is relatively new, we opt for a shorter review period. This allows us to course correct quickly if needed.
  2. Stability and Effectiveness: Policies that have remained unchanged for a considerable period and have proven to be effective can have longer review periods. This prevents unnecessary time spent on policies that are working well.
3. Integration into Regular Meetings: Policy reviews are treated as regular agenda items during our team meetings. They are discussed and evaluated alongside other topics of importance.
4. Review Process: Policy reviews are conducted using a round-based approach. Each team member has the opportunity to provide feedback and raise any concerns or suggestions regarding the policy under review.
5. Regular Review Cadence: We review our policies at regular intervals to proactively identify any issues before they escalate into urgent problems. By conducting these reviews systematically, we ensure that our policies remain up to date and aligned with our evolving needs.

By adhering to this policy review process, we aim to maintain a high standard of quality and relevance in our style team policies.

```

Reviewer 3: > 4. **Review Process**: Policy reviews are conducted using a round-based approach. Each team member has the opportunity to provide feedback and raise any concerns or suggestions regarding the policy under review.

I don't think this meta-policy on regular policy review is the right place to discuss how the discussion in the policy review should be facilitated. I also think we should be very careful about the degree of constraint we place on the facilitator for how they can most effectively facilitate discussion. In this context, I would prefer to state a goal, rather than a mechanism:

**Review Process**: Policy reviews should use a discussion approach that affords each team member the opportunity to provide feedback and raise any concerns or suggestions regarding the policy under review.

Reviewer 3: Also, I'd like to include a point here about whether there's a default assumption of "keep the policy" or "drop the policy", since that seems like a major rationale for policy reviews:

**Default disposition**: In organizations without a policy review process, policies often remain in effect by default, until explicitly rescinded. The policy review process exists in part to reverse that default disposition: policies require regular review and must be renewed in order to remain in effect. (Such review and renewal still takes all of these other factors into account, such as the degree of previously established confidence in the policy, and its stability over time.)

Author: @joshtriplett I went ahead and updated the proposed wording to include your suggestions. I didn't remove the recommendation to use rounds, but I did change it to be clearly a recommendation and to first lead with the core goals of encouraging everyone to provide feedback and creating an environment where everyone is heard equally.

Reviewer 3: @yaahc This solves most of my concern. My remaining concern with this new wording: we should add something to the prompt to invite people to raise issues with the policy other than just its efficacy at its intended purpose. (e.g. "this isn't the right purpose" or "this is causing other problems that aren't worth the tradeoff"...)

For instance, "Are there other issues with the policy that we should consider?"

## Operational Roles

The style team has various operational roles to delegate related and ongoing work to specific individuals, to clarify on our operations, and to coordinate sharing feedback with the members in those roles to enable iterative self improvement.

### Operational Lead

* Current Holder: [@calebcartwright](https://github.com/calebcartwright)
* Next Feedback Date: TODO

In order to stay in touch with where we want to be heading in the future, we need leadership. A team leader is paying attention to the team’s operations in relation to the team’s aim. What needs to be done, who agreed to do it. What is in the future to decide?

> 

Author: Just ran into this article and thought it might be a good one to save as recommended reading for anyone who takes on this role:

```suggestion

#### Recommended Reading

* https://www.sociocracyforall.org/leadership-without-hierarchy/

```

Responsibilities:

* Facilitates team meetings.
* Pays attention to equivalence[^5] during meetings.
* Supports planning of the agenda.
* Distinguishes facilitator voice from team member voice.

### Secretary

* Current Holder: [@calebcartwright](https://github.com/calebcartwright)
* Next Feedback Date: TODO

In order to manage continuity with the team’s past, we need to have written records. The secretary manages the notes during the meeting, makes sure the minutes are distributed and accessible. The secretary also manages the records of the team and is the interpreter of policies.

Responsibilities:

* Makes sure meeting minutes are taken, approved, and stored.
* Keeps track of all documents of the team.
* Interprets the meeting minutes in case of disagreement.
* Supports planning the agenda from the backlog.
    * Tracking when policies and roles are due for review.

## Meetings

> The style team meets weekly on wednesdays at 12:30pm PST [(everytimezone link)](https://everytimezone.com/s/3f88a253) for a weekly video sync. Agendas are posted in the [#T-style] zulip stream. 

Reviewer 3: ```suggestion
The style team meets weekly on Wednesdays at 12:30pm PST [(everytimezone link)](https://everytimezone.com/s/3f88a253) for a weekly video sync. Agendas are posted in the [#T-style] zulip stream. 
```

Unless otherwise noted, all of our meetings are public and open for anyone to attend. You will find the timing and event details on our [style team calendar](https://calendar.google.com/calendar/embed?src=d0564ed914a41cf4915bd5ebe6e2e4ec0ee1293fdc1d09d6f5bdb27d4f91c083%40group.calendar.google.com&ctz=America%2FLos_Angeles). We publish notes and minutes in written form in this github repository.

### Agenda, Backlog, and Minutes

> The style team stores its agendas, backlog, and meeting minutes in a single live [hackmd document](https://hackmd.io/@yaah/B1GZrzv4j) (rotating as necessary when the documents reach their length limit).

Reviewer 3: I think if we're going to make this public then we need to ensure that it isn't editable by non-members. Otherwise, someone's going to come by and blank it or scribble in it.

Author: we already share the link in zulip, I think I'd rather deal with this when it becomes an issue vs before, assuming they cannot vandalize it so hard as to destroy the entire git history pre-griefing

The agenda should be prepared in advanced of each meeting by the facilitator with the assistance of the scribe and the lead. The lead is responsible for looking forward and adding agenda items such as new requests from other teams and users or new priorities and goals. The scribe is responsible for looking backwards and adding agenda items from existing work such as items from the backlog or policies that are due for review. The agenda proposal is then presented at the beginning of each meeting by the facilitator for the rest of the team to consent or object too.

Consent not needed to add to items to the backlog, anybody is welcome to add items to the backlog, but consent needed to move from backlog to agenda (since agendas themselves require consent). An item being present on the backlog does not represent a commitment by the style team.

> 

Author: ```suggestion
#### Interrupting the Agenda

The style team acknowledges that urgent needs may arise during the meeting, and in order to address these needs, participants may interrupt the meeting by speaking up and immediately saying "Point of Order." This signals to the meeting facilitator and other participants that the following point is not an out of order interruption, but rather an urgent need that must take priority. This tool is particularly intended to allow the scribe to interrupt the meeting when they have unmet needs in their role as minutes taker, but it is available for any team member to use when they feel they have the need.

The meeting facilitator will then pause the current discussion and allow the participant with the point of order to state their concern. After the urgent matter is resolved, the facilitator will resume the previous discussion or return to the agenda. This policy is intended to ensure that the team can address urgent issues in a timely manner without interrupting the flow of the meeting unnecessarily.

```

Reviewer 3: > This tool is particularly intended to allow the scribe to interrupt the meeting when they have unmet needs in their role as minutes taker, but it is available for any team member to use when they feel it is necessary.

I would change this to:

"This tool is designed for meta-level concerns that take priority over the current discussion, such as the scribe interrupting the meeting to address issues preventing the recording of minutes (and stopping the discussion until they can resume recording minutes), or a team member interrupting a discussion to raise questions of time bounds or team scope."

> The meeting facilitator will then pause the current discussion and allow the participant with the point of order to state their concern.

"The meeting facilitator will then pause the current discussion, allow the participant raising the point of order to state their concern, and allow the team to address that point of order."

We identify the desired outcome in advance to avoid aimless discussions with unclear goals. The desired outcome can change during discussion as new information becomes available. It is the facilitator's responsibility to notice when this happens and formally make the change in the desired outcome with consent from the rest of the team. Extra clarity can be gained if, for every agenda item, we end by measuring whether we have achieved the desired outcome. Facilitators can make it a habit to pause before moving to a new agenda item by assessing whether the desired outcome has been achieved and by asking the secretary to read out loud what has been written in the notes.

The agenda backlog and minutes document is structured according to the following template:

```md
# T-style Minutes

> Meeting Link: https://meet.jit.si/t-style-style-team-task-force

Reviewer 3: We should not make this public, or if we do then it needs to be locked and require approval.

Author: not sure I agree but I'm happy to remove it for now pending that discussion

## Action Items

### Pending

* owner: bullet point list of items that are in progress and assigned to a specific person

### Completed

- [ ] owner: check list of items that are completed or assumed to be complete
- [x] owner: items are checked off once they've been reviewed in a meeting, confirmed complete, and given any relevant final status updates.
- [x] owner: after a meeting the items that were checked off are moved into the `#### Completed Action Items` section of the meeting they were reviewed in by the scribe

> 

Reviewer 3: ```suggestion
### Granularity

Action items should not generally list individual feedback/response comments within GitHub threads. An action item should generally be at the level of a GitHub PR, and people can look at the GitHub PR to find out its current status. That status may affect whether the action item is blocked, or who needs to look at it, or whether there are aspects of the action item to discuss in a meeting.
```

* “What worked well in the meeting?”
* “What could be improved in future meetings?”
* “Is there anything you are carrying out of the meeting that you’d like to get off your chest now?”

Meeting evaluations are an opportunity to learn from our meetings. We can either talk about content, process, or interpersonal dynamics. We utilize meeting evaluations to help ourselves inhabit a growth mindset. Our goal is to have meetings which are well-run, refreshing, connecting, and energizing. We achieve this goal by giving space for people to speak up about both the positive and negative aspects of how we're working together so that we can prioritize continuous improvement and positive connections.

Meeting check-outs are considered private and internal to the style team and are not recorded as part of our minutes. Exceptions to this can be made via an operational consent decision by the team, and are often useful in cases such as when new backlog or action items come up during the check-out.

> [Style Guide]: https://github.com/rust-lang/fmt-rfcs/blob/master/guide/guide.md

Reviewer 3: This link doesn't appear to be used anywhere.

```suggestion
```

> [#T-style]: https://rust-lang.zulipchat.com/#narrow/stream/346005-t-style

Reviewer 3: This appears to be referenced in only one place, and I don't think that place needs it to be a link. Can we drop this?

```suggestion
```
